d2/src/main/java/com/linkedin/d2/balancer/clients/BackupRequestsClient.java

```diff
@@ -195,24 +247,65 @@ public void restRequest(final RestRequest request, final RequestContext requestC
    */
   private void updateIfNeeded(String serviceName)
   {
-    List<Map<String, Object>> existing = _configs.get(serviceName);
     try
     {
       ServiceProperties serviceProperties = _loadBalancer.getLoadBalancedServiceProperties(serviceName);
```

Comment:
This is still calling the sync implementation, right?

Reply:
I created another method updatedIfNeeded(serviceName, callback) and introduced a configuration. If async is enabled, this path won't be invoked, the async path will be invoked instead. I did not remove the blocking path in the code, the control is from the config.

Reply:
Got it.